handlers.go

```diff
@@ -0,0 +1,304 @@
+package main
+
+import (
+	"Chirpy/domain"
+	"Chirpy/internal/auth"
+	"Chirpy/internal/database"
+	"Chirpy/tokens"
+
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"sort"
+	"time"
+
+	"github.com/google/uuid"
+)
+
+func (cfg *apiConfig) webhooksHandler(w http.ResponseWriter, r *http.Request) {
+	var webhookData domain.Webhook
+	if !checkApiKey(cfg, w, r) {
+		return
+	}
+	defer r.Body.Close()
+	err := json.NewDecoder(r.Body).Decode(&webhookData)
+	if err != nil {
+		cfg.respondWithError(w, http.StatusBadRequest, "Error decoding webhook")
+		return
+	}
+	if webhookData.Event != "user.upgraded" {
+		w.WriteHeader(http.StatusNoContent)
+		return
+	}
+
+	if webhookData.Data.UserID == uuid.Nil {
+		cfg.respondWithError(w, http.StatusBadRequest, "Invalid user ID")
+		return
+	}
+
+	_, err = cfg.db.CheckUser(r.Context(), webhookData.Data.UserID)
+	if err != nil {
+		cfg.respondWithError(w, http.StatusNotFound, "User Not Found")
+		return
+	}
+
+	_, err = cfg.db.UpdateChirpyRed(r.Context(), webhookData.Data.UserID)
+	if err != nil {
+		cfg.respondWithError(w, http.StatusInternalServerError, "Could not update Chirpy Red")
+		return
+	}
+
+	w.WriteHeader(http.StatusNoContent)
+}
+
+func (cfg *apiConfig) deleteChirpHandler(w http.ResponseWriter, r *http.Request) {
+	userId, err := getUserId(cfg, r.Header)
+	if err != nil {
+		cfg.respondWithError(w, http.StatusUnauthorized, "Unauthorized")
+		return
+	}
+
+	chirpId := r.PathValue("chirpID")
+	context := r.Context()
+	chirpStruct, err := getOneChirp(cfg, chirpId, context)
+	if err != nil {
+		cfg.respondWithError(w, http.StatusBadRequest, fmt.Sprintf("%v", err))
+	}
+	if chirpStruct.ID == uuid.Nil {
+		return
+	}
+
+	if userId == chirpStruct.UserID {
+		err := cfg.db.DeleteChirp(r.Context(), chirpStruct.ID)
+		if err != nil {
+			cfg.respondWithError(w, http.StatusInternalServerError, "Failed to delete chirp")
+			return
+		}
+
+		w.WriteHeader(http.StatusNoContent)
+		return
+	}
+	cfg.respondWithError(w, http.StatusForbidden, "This chirp not your (no access)")
```

Comment:
```suggestion
	if userId != chirpStruct.UserID {
		cfg.respondWithError(w, http.StatusForbidden, "This chirp not your (no access)")
		return
	}
	
	err := cfg.db.DeleteChirp(r.Context(), chirpStruct.ID)
	if err != nil {
		cfg.respondWithError(w, http.StatusInternalServerError, "Failed to delete chirp")
		return
	}

	w.WriteHeader(http.StatusNoContent)
```

лучше сначала error path отработать, уменьшает вложенность